solidity/random-beacon/contracts/libraries/DKG.sol

```diff
@@ -148,24 +150,40 @@ library DKG {
     {
         state = State.IDLE;
 
-        if (self.startBlock > 0) {
-            state = State.KEY_GENERATION;
+        if (self.sortitionPool.isLocked()) {
+            state = State.WARMED_UP;
 
-            if (block.number > self.startBlock + offchainDkgTime) {
-                state = State.AWAITING_RESULT;
+            if (self.startBlock > 0) {
+                state = State.KEY_GENERATION;
 
-                if (self.submittedResultBlock > 0) {
-                    state = State.CHALLENGE;
+                if (block.number > self.startBlock + offchainDkgTime) {
+                    state = State.AWAITING_RESULT;
+
+                    if (self.submittedResultBlock > 0) {
+                        state = State.CHALLENGE;
+                    }
                 }
             }
         }
     }
 
-    function start(Data storage self, uint256 seed) internal {
+    /// @notice Warms up the group creation process by locking the sortition
+    ///         pool.
+    function warmUp(Data storage self) internal {
```

Comment:
Maybe `lockState()` or `lockSortitionPool()`? I think this would be easier to follow from the perspective of `RandomBeacon` contract. That contract also uses the pool (join / update operator), so being explicit that it's about to be locked makes sense.

Reply:
Yeah, good idea. I had no idea for a good nomenclature here. Thanks for the inspiration! Will change as you proposed.

Reply:
See https://github.com/keep-network/keep-core/pull/2691/commits/a29a5ad1f24125d53b37cef20129d6c5977764be